Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Operator/FleetManagement.hs

```diff
@@ -213,9 +214,13 @@ postFleetManagementFleetLinkSendOtp merchantShortId opCity requestorId req = do
         personData <- DRegistration.createFleetOwnerDetails personAuth merchant.id merchantOpCityId True deploymentVersion.getDeploymentVersion enabled
         pure personData
 
-  SA.checkForFleetAssociationOverwrite merchant fleetOwner.id
-  checkAssocOperator <- B.runInReplica $ QFOA.findByFleetOwnerIdAndOperatorId fleetOwner.id operator.id True
-  when (isJust checkAssocOperator) $ throwError (InvalidRequest "Fleet already associated with operator")
+  existingFOAssociations <- QFOA.findAllByFleetOwnerId fleetOwner.id True
+  when (merchant.overwriteAssociation /= Just True && notNull existingFOAssociations)
+    . throwError
+    $ InvalidRequest "Fleet already associated with another operator"
+  when (any (\foa -> Id foa.fleetOwnerId == fleetOwner.id && Id foa.operatorId == operator.id) existingFOAssociations)
+    . throwError
+    $ InvalidRequest "Fleet already associated with operator"
```

Comment:
Could you move either this logic to `checkForFleetAssociationOverwrite` function, or remove this function at all?